src/components/DataViewer/DataViewer.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import React from "react";
+import styled from "styled-components";
+import Modal from "react-modal";
+
+interface DataViewerProps {
+    data: { [key: string]: string | number | null };
+    title: string;
+    isOpen: boolean;
+    onRequestClose: () => void;
+}
+
+const ModalWithStyle = styled(Modal)`
+    position: absolute;
+    inset: 50% 40px 40px 50%;
```

Comment:
Try and use max or min on the inset property here to try and make it responsive, hopefully that works

Reply:
Seems like max and min isn't necessary to make it responsive (?) but we removed those px-dependent number and stick to percentage, and it is responsive.